app/auto_responder.go

```diff
@@ -43,11 +43,21 @@ func (a *App) SendAutoResponse(channel *model.Channel, receiver *model.User) (bo
 		return false, nil
 	}
 
+	rootID := post.Id
+	if post.RootId != "" {
+		rootID = post.RootId
+	}
+
+	parentID := post.Id
+	if post.ParentId != "" {
+		parentID = post.ParentId
+	}
```

Comment:
According to https://community-daily.mattermost.com/core/pl/np785upq8pgujcag478cdyjs6o `ParentId` is deprecated. Should it still get used here?

Reply:
I know it is deprecated, but it is still there. For the sake of consistency, I preferred to check both. But if you prefer, I can remove this piece of code and use only RootID

Reply:
I'm 3/5 that new code shouldn't use a deprecated field any longer.